src/SourceBuild/content/test/Microsoft.DotNet.UnifiedBuild.Tests/Microsoft.DotNet.UnifiedBuild.Tests.csproj

```diff
@@ -46,6 +49,12 @@
       <RuntimeHostConfigurationOption Include="Microsoft.DotNet.UnifiedBuild.Tests.NoDiagnosticMessages">
         <Value>$(VSTestUseMSBuildOutput)</Value>
       </RuntimeHostConfigurationOption>
+      <RuntimeHostConfigurationOption Include="Microsoft.DotNet.UnifiedBuild.Tests.Packages">
+        <Value>@(_ProducedPackages->'%(Identity)')</Value>
+      </RuntimeHostConfigurationOption>
+      <RuntimeHostConfigurationOption Include="Microsoft.DotNet.UnifiedBuild.Tests.GitInfoDir">
+        <Value>$(GitInfoDir)</Value>
+      </RuntimeHostConfigurationOption>
```

Comment:
What do you need the .props files from the git info dir for? That directory will eventually be deleted when we get to the versioning aspect of the VMR.

Reply:
The build asset manifest is the right data source for that. It lists all the packages and its versions.

Reply:
Where is the build asset manifest?

Reply:
https://github.com/dotnet/dotnet/blob/3de4f5f0f54de33d9ced357ad0cb7a1ef5982e1b/build.proj#L32

Reply:
To see an example how it looks, check the pipeline artifacts of a build.

Reply:
It looks like I'd have to move the test target to the outer `dotnet.proj` build to run it after the asset manifest is created. I'll do that in a follow up and just use the versions in each package's nuspec for now to keep this PR small.

Reply:
Yes, that would be necessary. I think we should wait with this PR until https://github.com/dotnet/installer/pull/19290 goes in anyway as that will significantly change when and how we sequence tests into the build.

Reply:
Sounds good, I merged that PR into this branch so we can test the combination of the two.

Reply:
My PR is now in main. As mentioned previously, please use the build manifest as an input now. You can extract a version from it via the `XmlPeek` msbuild task. Here's an example: https://github.com/dotnet/installer/blob/12eec080d5008c56731624c6a8fb9fe2960a1ca0/src/SourceBuild/content/repo-projects/Directory.Build.targets#L462-L465

Reply:
I've been using the versions in the nuspec of the built packages, will the build manifest and nuspec versions ever differ?

Reply:
No, those versions are in sync. Do you just need the version from the nuspec or anything else? If it's just the version then please really consider using the vertical asset manifest. See https://github.com/dotnet/source-build/issues/4302 which lists further reasons to depend on the asset manifest file.

Reply:
I use the nuspec for the package full name as well, but I could start from the xml and find a package with a matching file name. Will the vertical asset manifest have full versions when we switch to producing -dev versioned packages?

Reply:
When we switch to -dev versioned packages, both the package name and the asset manifest information will have the "9.0.0-dev" versions.

Reply:
If you want to get a set of MSBuild items for "package id + version" for all package in an asset manifest, you can also use the model I did in https://github.com/dotnet/installer/pull/19389 in the `GetProducedPackages` target and dependencies.

Reply:
I'm OK with changing the source-of-truth to the merged vertical asset manifest in a follow-up. Please file a tracking issue.